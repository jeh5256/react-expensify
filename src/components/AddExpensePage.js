import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import { dispatch } from 'rxjs/internal/observable/pairs';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const matchDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, matchDispatchToProps)(AddExpensePage);