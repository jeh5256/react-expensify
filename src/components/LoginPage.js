import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify App</h1>
            <p>The modern way to keep track of daily expenses</p>
            <button className="button" onClick={startLogin}>Login With Google</button>
        </div>
    </div>
);

const matchDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, matchDispatchToProps)(LoginPage);