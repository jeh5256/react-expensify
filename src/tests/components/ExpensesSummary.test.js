import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should rended ExpensesSummery with one expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={150} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should rended ExpensesSummery with one expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={1200} />);
    expect(wrapper).toMatchSnapshot();
});