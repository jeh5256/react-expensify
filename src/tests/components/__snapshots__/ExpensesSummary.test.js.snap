// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should rended ExpensesSummery with one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.50
  </h1>
</div>
`;

exports[`Should rended ExpensesSummery with one expense correctly 2`] = `
<div>
  <h1>
    Viewing 
    5
     
    expenses
     totalling 
    $12.00
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $235,123,409.87
  </h1>
</div>
`;
